doc/NPCs.md

```diff
@@ -0,0 +1,251 @@
+
+This example adds the "I'm going now!" to the all the listed topics.
+```JSON
+{
+    "type": "talk_topic",
+    "id": [ "TALK_ARSONIST", "TALK_STRANGER_FRIENDLY", "TALK_STRANGER_NEUTRAL" ],
+    "dynamic_line": "What now?",
+    "responses": [
+        {
+            "text": "I'm going now.",
+            "topic": "TALK_DONE"
+        }
+    ]
+}
+```
+
+### dynamic_line
+This is the line spoken by the NPC. See the chapter for dynamic_line below. It is optional, if it's not defined and this topic has the same id as a built in topic, the `dynamic_line` from that built in topic will be used. Otherwise the NPC will say nothing.
+
+### response
+An array with possible responses, it must not be empty. Each entry must be a response object, see below.
+
+### replace_built_in_responses
+A boolean (optional, default is `false`) that defines whether to dismiss the build in responses for that topic. If there are no built in responses, this won't do anything. If `true`, the built in responses are ignored and only those from this definition in json are used. If `false`, the responses from json are used *and* the built in responses (if any).
+
+---
+
+## dynamic_line
+A dynamic line can either be a simple string, or an complex object, or an array with `dynamic_line` entries. If it's an array, an entry will be chosen randomly every time the NPC needs it. Each entry has the same probability.
+
+Example:
+```JSON
+"dynamic_line": [
+    "generic text",
+    {
+        "npc_female": [ "text1", "text2", "text3" ],
+        "npc_male": { "u_female": "text a", "u_male": "text b" }
+    }
+]
+```
+
+A complex `dynamic_line` usually contains several `dynamic_line` entry and some condition that determines which is used. They can be defined like this:
+
+### Based on the gender of the NPC / NPC
+The dynamic line is chosen based on the gender of the NPC, both entries must exists. Both entries are parsed as `dynamic_line`.
+
+```JSON
+{
+    "npc_male": "I'm a man.",
+    "npc_female": "I'm a woman."
+}
+```
+
+### Based on the gender of the player character
+The dynamic line is chosen based on the gender of the player character, both entries must exists. Both entries are parsed as `dynamic_line`. `u` is the player character.
+
+```JSON
+{
+    "u_male": "You're a man.",
+    "u_female": "You're a woman."
+}
+```
+
+### Based on items worn by the player character
+The dynamic line is chosen based on the gender of the player character, both entries are optional, but you should make sure the NPC says something all the time. Both entries are parsed as dynamic_line. `u` is the player character. The `u_is_wearing` string should be a valid item id.
```

Comment:
Hmm, surely this dynamic line is based on the PC's equipped gear?  And I'm glad you like the ears.  :-3